src/map_field.cpp

```diff
@@ -1719,54 +1672,86 @@ void map::player_in_field( player &u )
 
 void map::creature_in_field( Creature &critter )
 {
+    bool inside = false;
+    player *u = critter.as_player();
+    field &cur_field = get_field( critter.pos() );
     if( critter.is_monster() ) {
-        monster_in_field( *static_cast<monster *>( &critter ) );
-    } else if( player *p = critter.as_player() ) {
-        player_in_field( *p );
+        monster_in_field( *static_cast<monster *>( &critter ), cur_field, inside );
+    } else {
+        if( u && u->in_vehicle ) {
+            // If we are in a vehicle figure out if we are inside (reduces effects usually)
+            // and what part of the vehicle we need to deal with.
+            if( u->in_vehicle ) {
```

Comment:
You already checked this, see a few lines above. It is always `true` at that point.